lib/active_shipping/carrier.rb

```diff
@@ -1,7 +1,22 @@
 module ActiveShipping
+
+  # Carrier is abstract the base class for all supported carriers.
+  #
+  # To implement support for a carrier, you should subclass this class and
+  # implement all the methods the carrier supports.
+  #
+  # @see #find_rates
+  # @see #create_shipment
```

Comment:
Yeah, some implement it to return a checking number only. And some don't implement it at all. This is why it says "you should subclass this class and implement all the methods the carrier supports." The `find_tracking_info` method was previously not included in the base class at all, but is implemented by many carriers.

The abstraction is pretty poorly defined (a trait shared with the other abstraction layer gems we have)